mpc-core/src/ot/extension/sender.rs

```diff
@@ -154,21 +175,47 @@ where
         &mut self,
         base_sender_setup: BaseSenderSetup,
     ) -> Result<BaseReceiverSetup, ExtSenderCoreError> {
-        Ok(self.base.setup(&self.base_choice, base_sender_setup)?)
+        if self.state != State::Initialized {
+            return Err(ExtSenderCoreError::WrongState);
+        }
+        self.receiver_cointoss_commit = Some(base_sender_setup.cointoss_commit);
+        self.state = State::BaseSetup;
+        Ok(BaseReceiverSetup {
+            setup: self
+                .base
+                .setup(&self.base_choice, base_sender_setup.setup)?,
+            cointoss_share: self.cointoss_share,
+        })
     }
 
     fn base_receive(&mut self, payload: BaseSenderPayload) -> Result<(), ExtSenderCoreError> {
-        let receive = self.base.receive(payload)?;
+        if self.state != State::BaseSetup {
+            return Err(ExtSenderCoreError::WrongState);
+        }
+        let receive = self.base.receive(payload.payload)?;
         self.set_seeds(receive);
-        self.state = State::BaseSetup;
+
+        // check the decommitment for the other party's share
+        if sha256(&payload.cointoss_share)
+            != self
```

Comment:
This should probably be a constant time equality check, no? Use `subtle::ConstantTimeEq` for this. The expression should be something like
```rust
if !bool::from(sha256(&payload.cointoss_share).ct_eq(other_value)) {
   // ...
}
```

Reply:
You should check the rest of the code for non-constant-time equalities. I'll probably miss some